lib/action_subscriber/railtie.rb

```diff
@@ -3,10 +3,11 @@ class Railtie < ::Rails::Railtie
     config.action_subscriber = ::ActionSubscriber.config
 
     ::ActiveSupport.on_load(:active_record) do
+      require "action_subscriber/middleware/active_record/connection_management_async"
       require "action_subscriber/middleware/active_record/connection_management"
       require "action_subscriber/middleware/active_record/query_cache"
 
-      ::ActionSubscriber.config.middleware.use ::ActionSubscriber::Middleware::ActiveRecord::ConnectionManagement
+      ::ActionSubscriber.config.middleware.use ::ActionSubscriber::Middleware::ActiveRecord::ConnectionManagementAsync
```

Comment:
What if we just replaced the functionality in the existing middleware object (keep the same name). I don't know of any cases in which this will break for people using the library. It is only a performance improvement.

I'm okay with doing a major version bump or just replacing the existing class with the new implementation :+1: